src/bin/server/main.rs

```diff
@@ -0,0 +1,75 @@
+use std::{
+    io::{self, Read},
+    net::SocketAddr,
+    process,
+};
+
+use snafu::{ResultExt, Snafu};
+use structopt::StructOpt;
+use tonic::transport::Server;
+
+mod config;
+mod rpc;
+
+#[derive(Debug, StructOpt)]
+enum Commands {
+    Config(CommandsConfig),
+    Run,
+}
+
+#[derive(Debug, StructOpt)]
+enum CommandsConfig {
+    New { address: SocketAddr },
+}
+
+#[derive(Debug, Snafu)]
+enum Error {
+    ConfigEncode { source: toml::ser::Error },
+    RunReadConfig { source: io::Error },
+    RunDecodeConfig { source: toml::de::Error },
+    RunServer { source: tonic::transport::Error },
+}
+
+fn config(cmd: CommandsConfig) -> Result<(), Error> {
+    match cmd {
+        CommandsConfig::New { address } => {
+            let config = config::Config { address };
+            let encoded = toml::to_string_pretty(&config).context(ConfigEncode)?;
+
+            println!("{}", encoded);
```

Comment:
Consider using the tracing crate
```suggestion
            tracing::debug!("{}", encoded);
```

Reply:
In this case, it is intended, as the config is printed on stdout.
btw `println` indeed looks to me as a "debug" statement, but it seems the normal way to print smth to stdout, WDYT?

Reply:
stdout may be lost when running as a daemon, whereas logging macros do select an output themselves and depend on the configured logger (which may just log everything to stdout, but gives more flexibility).